src/lib/utils.ts

```diff
@@ -8,7 +8,7 @@ import os from "os";
 import path from "path";
 import request from "superagent";
 
-export type TempDownloadHandler = <T>(filePath: string) => Promise<T> | T;
+export type TempDownloadHandler = (filePath: string) => Promise<any> | any;
```

Comment:
Is this change supposed to be part of this PR?

Reply:
This change is part of this PR. I started fixing small type issues I could find. Then thought I'd just get this PR'd first, and can start adding types as time goes on. The longer this split would sit in flux, the longer it would cause pain/disruptions.